internal/integrationloader/git.go

```diff
@@ -0,0 +1,147 @@
+	return loadChangelog(l)
+}
+
+func (l GitLoader) LoadImages() (Images, error) {
+	images := Images{}
+	imagesPath := path.Join(l.integrationPath, defaultImagesDirName)
+
+	// attempt to resolve the git ref to a revision
+	hash, err := l.repo.ResolveRevision(plumbing.Revision(l.ref))
+	if err != nil {
+		return images, fmt.Errorf("error resolving git revision %s: %w", l.ref, err)
+	}
+
+	// attempt to retrieve the commit object for the hash
+	commit, err := l.repo.CommitObject(*hash)
+	if err != nil {
+		return images, fmt.Errorf("error retrieving commit for hash %s: %w", hash, err)
+	}
+
+	// attempt to retrieve the directory tree for the commit
+	tree, err := commit.Tree()
+	if err != nil {
+		return images, fmt.Errorf("error retrieving tree for commit %s: %w", hash, err)
+	}
+
+	// attempt to load the file from the tree
+	// file, err := tree.FindEntry(imagesPath)
+	// if err != nil {
+	// 	if errors.Is(err, plumbingobject.ErrEntryNotFound) {
+	// 		return images, nil
+	// 	}
+	// 	return images, fmt.Errorf("error accessing %s for hash %s: %w", imagesPath, hash, err)
+	// }
+
+	imagesTree, err := tree.Tree(imagesPath)
+	if err != nil {
+		if errors.Is(err, object.ErrDirectoryNotFound) {
+			return images, nil
+		}
+		return images, err
+	}
+
+	if err := imagesTree.Files().ForEach(func(f *object.File) error {
+		match, _ := regexp.MatchString(reImageExtensions, f.Name)
+		if match {
+			data, err := f.Contents()
+			if err != nil {
+				return err
+			}
+			images[f.Name] = data
+		}
+		return nil
+	}); err != nil {
+		return images, err
+	}
+
+	return images, nil
+}
+
+func (l GitLoader) LoadLogo() (string, error) {
+	return loadLogo(l)
+}
+
+func (l GitLoader) LoadReadme() (string, error) {
+	return loadReadme(l)
+}
+
+func (l GitLoader) LoadResources() (string, error) {
+	return loadResources(l)
+}
+
+func (l GitLoader) GetFileContentsAsBytes(relativePath string) ([]byte, error) {
+	contents, err := l.GetFileContentsAsString(relativePath)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(contents), nil
+}
+
+func (l GitLoader) GetFileContentsAsString(relativePath string) (string, error) {
+	filePath := path.Join(l.integrationPath, relativePath)
+
+	// attempt to resolve the git ref to a revision
+	hash, err := l.repo.ResolveRevision(plumbing.Revision(l.ref))
+	if err != nil {
+		return "", fmt.Errorf("error resolving git revision %s: %w", l.ref, err)
+	}
+
+	// attempt to retrieve the commit object for the hash
+	commit, err := l.repo.CommitObject(*hash)
+	if err != nil {
+		return "", fmt.Errorf("error retrieving commit for hash %s: %w", hash, err)
+	}
+
+	// attempt to retrieve the directory tree for the commit
+	tree, err := commit.Tree()
+	if err != nil {
+		return "", fmt.Errorf("error retrieving tree for commit %s: %w", hash, err)
+	}
+
+	// attempt to load the file from the tree
+	file, err := tree.File(filePath)
+	if err != nil {
+		return "", fmt.Errorf("error accessing %s for hash %s: %w", filePath, hash, err)
+	}
```

Comment:
it's not clear to me why, but when attempting to run `generate` on this branch I'm seeing:

```bash
$ go run main.go catalog generate --repo-dir ../catalog                          validate 
7:48AM INF Found integration version name=ansible-tower-remediation namespace=ansible tag=ansible/ansible-tower-remediation/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=cassandra-monitoring namespace=cassandra tag=cassandra/cassandra-monitoring/20220217.0.0 version=20220217.0.0
7:48AM INF Found integration version name=helloworld namespace=example tag=example/helloworld/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=kubernetes-monitoring namespace=kubernetes tag=kubernetes/kubernetes-monitoring/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=mattermost-alerts namespace=mattermost tag=mattermost/mattermost-alerts/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=nginx-monitoring namespace=nginx tag=nginx/nginx-monitoring/20220217.0.0 version=20220217.0.0
7:48AM INF Found integration version name=pagerduty-incidents namespace=pagerduty tag=pagerduty/pagerduty-incidents/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=slack-alerts namespace=slack tag=slack/slack-alerts/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=sumologic-analytics namespace=sumologic tag=sumologic/sumologic-analytics/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=disk-monitoring namespace=system tag=system/disk-monitoring/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=host-monitoring namespace=system tag=system/host-monitoring/20220223.0.0 version=20220223.0.0
7:48AM INF Found integration version name=network-interface-monitoring namespace=system tag=system/network-interface-monitoring/20220217.0.0 version=20220217.0.0
namespace: system, name: disk-monitoring, version: 20220223.0.0
7:48AM ERR Failed to process integration version error="error accessing ../catalog/integrations/system/disk-monitoring/sensu-integration.yaml for hash c75486b929eee43e96ee70eb7006f0519ea71096: file not found" integration=disk-monitoring namespace=system version=20220223.0.0
error: error processing catalog: error processing integration version: error accessing ../catalog/integrations/system/disk-monitoring/sensu-integration.yaml for hash c75486b929eee43e96ee70eb7006f0519ea71096: file not found
exit status 1
```

Reply:
Interestingly it doesn't fail with validation subcommand.

Reply:
It also seems like it exists? https://github.com/sensu/catalog/blob/c75486b929eee43e96ee70eb7006f0519ea71096/integrations/system/disk-monitoring/sensu-resources.yaml